activerecord/test/cases/adapter_test.rb

```diff
@@ -238,6 +238,14 @@ def test_log_invalid_encoding
         end
       end
     end
+
+    def test_tables_is_deprecated
+      if current_adapter?(:MysqlAdapter, :Mysql2Adapter, :SQLite3Adapter)
+        assert_deprecated { @connection.tables }
+      elsif current_adapter?(:PostgreSQLAdapter)
+        assert_deprecated { @connection.tables(:books) }
```

Comment:
Let's split this into a separate test-case. Passing arguments is not just deprecated for the PostgreSQL adapter:

```ruby
    if current_adapter?(:MysqlAdapter, :Mysql2Adapter, :SQLite3Adapter)
      def test_tables_returning_both_tables_and_views_is_deprecated
        assert_deprecated { @connection.tables }
      end
    end

    def test_passing_arguments_to_tables_is_deprecated
      assert_deprecated { @connection.tables(:books) }
    end
```